src/EditorFeatures/CSharpTest/ForeachToFor/ForeachToForTests.cs

```diff
@@ -0,0 +1,1177 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.ForeachToFor;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ForeachToFor
+{
+    public partial class ForeachToForTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new CSharpForeachToForCodeRefactoringProvider();
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task EmptyBlockBody()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach[||](var a in array)
+        {
+        }
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < array.Length; i++)
```

Comment:
is post-increment what we want here?

Reply:
also, more important: does this respect var/explicit type preferences? what if someone prefers explicit type on predefined types, therefore on int as an index, but prefered var for the element type

Reply:
for now, I made it to always use "var". didn't spend time to see how other feature did. if there is common code around this, I will follow those.

Reply:
> "is post-increment what we want here?" 
what style are you prefer?

Reply:
Keep post increment.  It is idiomatic and appropriate for C#.

Reply:
Shouldn't we just carry whatever type was spelled out in the `for` statement? 
If you have `for (int i ...` then we'd produce `foreach (int i in ...` and if you have `for (var i ...` then we'd produce `foreach (var i in ...`.

---
In reply to: [174340427](https://github.com/dotnet/roslyn/pull/25460#discussion_r174340427) [](ancestors = 174340427)

Reply:
> If you have for (int i ... then we'd produce foreach (int i in ...

No.  These types are unrelated.  one is the type of the index (normally an int).  The other is the type that that the iterator returns (something random).

Reply:
D'oh. My bad

---
In reply to: [174615970](https://github.com/dotnet/roslyn/pull/25460#discussion_r174615970) [](ancestors = 174615970)